src/sagemaker_mxnet_container/training.py

```diff
@@ -13,14 +13,69 @@
 from __future__ import absolute_import
 
 import logging
+import os
+import socket
+import subprocess
+
+from retrying import retry
 
 import sagemaker_containers.beta.framework as framework
 
+ROLES = ['worker', 'scheduler', 'server']
+
 logger = logging.getLogger(__name__)
 
 
+def _scheduler_host(hosts):
+    return hosts[0]
+
+
+def _env_vars_for_role(role, hosts, ps_port, ps_verbose):
+    if role in ROLES:
+        return {
+            'DMLC_NUM_WORKER': str(len(hosts)),
+            'DMLC_NUM_SERVER': str(len(hosts)),
+            'DMLC_ROLE': role,
+            'DMLC_PS_ROOT_URI': _host_lookup(_scheduler_host(hosts)),
+            'DMLC_PS_ROOT_PORT': ps_port,
+            'PS_VERBOSE': ps_verbose,
+        }
+
+    raise ValueError('Unexpected role: {}'.format(role))
+
+
+def _run_mxnet_process(role, hosts, ps_port, ps_verbose):
+    role_env = os.environ.copy()
+    role_env.update(_env_vars_for_role(role, hosts, ps_port, ps_verbose))
+    return subprocess.Popen("python -c 'import mxnet'", shell=True, env=role_env).pid
```

Comment:
We are not really using the pid anywhere. Also does this output anything? do we want to handle failures?

Reply:
nope. removed the return statement.